### PR TITLE
am: Implement exit locking and self exit commands

### PR DESCRIPTION
**Pull request by DarkLordZach**
This adds the ExitLock pair of commands, which on HW make the user wait/confirm they want to exit. Apps can use this to allow extra time (up to 15s) for cleanup tasks. In yuzu, this presents as an additional confirmation dialog.

This also adds the `Exit` Command, closing #2664 . This simply closes the current application.

### CODE REVIEW

**Comment by bunnei** [comment]
@DarkLordZach needs a rebase

File: src/yuzu/main.cpp [hunk 1]
```diff
@@ -2105,6 +2118,34 @@ bool GMainWindow::ConfirmChangeGame() {
     return answer != QMessageBox::No;
 }
 
+bool GMainWindow::ConfirmForceLockedExit() {
+    if (emu_thread == nullptr)
+        return true;
+
+    auto answer =
```
Comment (lioncash) [inline src/yuzu/main.cpp]:
```suggestion
    const auto answer =
```

File: src/yuzu/main.cpp [hunk 2]
```diff
@@ -2105,6 +2118,34 @@ bool GMainWindow::ConfirmChangeGame() {
     return answer != QMessageBox::No;
 }
 
+bool GMainWindow::ConfirmForceLockedExit() {
```
Comment (lioncash) [inline src/yuzu/main.cpp]: This can be a const member function

**Comment by MysticExile** [comment]
@DarkLordZach Needs a rebase

**Comment by Morph1984** [comment]
@DarkLordZach Reminder to rebase and address comments

**Comment by ogniK5377** [comment]
Build fail, please address when you can @DarkLordZach

File: src/yuzu/main.cpp (hunk 2 above)
Comment (DarkLordZach) [reply src/yuzu/main.cpp]: can't be because QMessageBox::question won't accept a const ptr as its first argument.

**Pull request merged**